### ISSUE TITLE
WINXP Zip Download Fails

### ISSUE THREAD

**Issue by joulesbeef** [c00w/bitHopper#599]
D:\Users\joulesbeef\Desktop\currentminer\bitHopper>python bitHopper.py
Traceback (most recent call last):
  File "bitHopper.py", line 34, in <module>
    import diff
  File "D:\Users\joulesbeef\Desktop\currentminer\bitHopper\diff.py", line 8, in <module>
    from btcnet_wrapper import btcnet_info
  File "D:\Users\joulesbeef\Desktop\currentminer\bitHopper\btcnet_wrapper.py", line 10, in <module>
    repo = Repo.init("btcnet_info")
  File "D:\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\repo\base.py", line 656, in init
    output = git.init(*_kwargs)
  File "D:\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py", line 227, in <lambda>
    return lambda *args, *_kwargs: self._call_process(name, _args, *_kwargs)
  File "D:\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py", line 456, in _call_process
    return self.execute(call, *__kwargs)
  File "D:\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py", line 335, in execute
    *_subprocess_kwargs
  File "D:\Python27\lib\subprocess.py", line 679, in __init__
    errread, errwrite)
  File "D:\Python27\lib\subprocess.py", line 893, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified

D:\Users\joulesbeef\Desktop\currentminer\bitHopper>

**Comment by c00w** [comment]
Shoot. I was hoping this would work cleanly in windows. I'll write a zip download version in case gitpython fails.

**Comment by c00w** [comment]
You need to install git: http://code.google.com/p/msysgit/downloads/list?can=3 I've added it to the readme.

**Comment by joulesbeef** [comment]
installed git off that link you gave me(already had tortoise git installed)
and i get the same error. :(..  win7 32bit

D:\Users\joulesbeef\Desktop\currentminer\bitHopper>python bitHopper.py
Traceback (most recent call last):
  File "bitHopper.py", line 34, in <module>
    import diff
  File "D:\Users\joulesbeef\Desktop\currentminer\bitHopper\diff.py", line
8, in <module>
    from btcnet_wrapper import btcnet_info
  File
"D:\Users\joulesbeef\Desktop\currentminer\bitHopper\btcnet_wrapper.py",
line 10, in <module>
    repo = Repo.init("btcnet_info")
  File
"D:\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\repo\base.py",
line 656, in init
    output = git.init(*_kwargs)
  File
"D:\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py",
line 227, in <lambda>
    return lambda *args, *_kwargs: self._call_process(name, _args, *_kwargs)
  File
"D:\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py",
line 456, in _call_process
    return self.execute(call, *__kwargs)
  File
"D:\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py",
line 335, in execute
    *_subprocess_kwargs
  File "D:\Python27\lib\subprocess.py", line 679, in __init__
    errread, errwrite)
  File "D:\Python27\lib\subprocess.py", line 893, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified

On Thu, Feb 9, 2012 at 10:02 PM, Colin Rice <
reply@reply.github.com

**Comment by macboy80** [comment]
I'm getting the same error, too. Followed new directions in README. Everything appeared to be successful.
Win7 x64 - Latest version from Git

logfile-errors.txt :

Traceback (most recent call last):
  File "bitHopper.py", line 34, in <module>
    import diff
  File "D:\BitHopper\bitHopper\diff.py", line 8, in <module>
    from btcnet_wrapper import btcnet_info
  File "D:\BitHopper\bitHopper\btcnet_wrapper.py", line 10, in <module>
    repo = Repo.init("btcnet_info")
  File "D:\BitHopper\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\repo\base.py", line 656, in init
    output = git.init(*_kwargs)
  File "D:\BitHopper\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py", line 227, in <lambda>
    return lambda *args, *_kwargs: self._call_process(name, _args, *_kwargs)
  File "D:\BitHopper\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py", line 456, in _call_process
    return self.execute(call, *__kwargs)
  File "D:\BitHopper\Python27\lib\site-packages\gitpython-0.3.2.rc1-py2.7.egg\git\cmd.py", line 335, in execute
    *_subprocess_kwargs
  File "D:\BitHopper\Python27\lib\subprocess.py", line 679, in __init__
    errread, errwrite)
  File "D:\BitHopper\Python27\lib\subprocess.py", line 893, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified

**Comment by c00w** [comment]
Shoot. Let me write a fallback downloader.

**Comment by c00w** [comment]
There is a fallback zip downloader. It works on my machine. Update and try again.

**Comment by macboy80** [comment]
Looks like it didn't get a chance to get to that error. I had this one come up on latest pull.

Traceback (most recent call last):
  File "bitHopper.py", line 34, in <module>
    import work
  File "D:\BitHopper\bitHopper\work.py", line 10, in <module>
    import httplib2
ImportError: No module named httplib2

**Comment by c00w** [comment]
You need to install httplib2. See the readme.

**Comment by macboy80** [comment]
Completely missed that update... It's installed now and getting this...

EDIT: Remove old error

Traceback (most recent call last):
  File "bitHopper.py", line 35, in <module>
    import diff
  File "D:\BitHopper\bitHopper\diff.py", line 9, in <module>
    from btcnet_wrapper import btcnet_info
  File "D:\BitHopper\bitHopper\btcnet_wrapper.py", line 79, in <module>
    Install_btcnet()
  File "D:\BitHopper\bitHopper\btcnet_wrapper.py", line 76, in Install_btcnet
    Pull_zip()
  File "D:\BitHopper\bitHopper\btcnet_wrapper.py", line 40, in Pull_zip
    headers, content = httplib2.Http(disable_ssl_certificate_validation=True).request('https://github.com/c00w/btcnet_info/zipball/master')
  File "D:\BitHopper\Python27\lib\site-packages\httplib2__init__.py", line 1445, in request
    (response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
  File "D:\BitHopper\Python27\lib\site-packages\httplib2__init__.py", line 1197, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "D:\BitHopper\Python27\lib\site-packages\httplib2__init__.py", line 1139, in _conn_request
    raise ServerNotFoundError("Unable to find the server at %s" % conn.host)
httplib2.ServerNotFoundError: Unable to find the server at github.com

**Comment by Clipseza** [comment]
Ive done some testing on windows for you, lag issues seems still exactly like previous updates since moving to gevent.

One other thing ive noticed in debug, rpcworklog seems to stall occasionally(might be due to writing to db?) which hangs the hopper completely then continues about 10sec later when it occurs.

**Comment by c00w** [comment]
@macboy80 github got hit with a ddos. Its back up and that should fix itself.

@Clipseza I'm not sure why rpcworklog would be lagging. I'll take a look.

**Comment by macboy80** [comment]
Still not grabbing the zip...

Traceback (most recent call last):
  File "bitHopper.py", line 35, in <module>
    import diff
  File "D:\BitHopper\bitHopper\diff.py", line 9, in <module>
    from btcnet_wrapper import btcnet_info
  File "D:\BitHopper\bitHopper\btcnet_wrapper.py", line 79, in <module>
    Install_btcnet()
  File "D:\BitHopper\bitHopper\btcnet_wrapper.py", line 76, in Install_btcnet
    Pull_zip()
  File "D:\BitHopper\bitHopper\btcnet_wrapper.py", line 40, in Pull_zip
    headers, content = httplib2.Http(disable_ssl_certificate_validation=True).request('https://github.com/c00w/btcnet_info/zipball/master')
  File "D:\BitHopper\Python27\lib\site-packages\httplib2__init__.py", line 1445, in request
    (response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
  File "D:\BitHopper\Python27\lib\site-packages\httplib2__init__.py", line 1197, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "D:\BitHopper\Python27\lib\site-packages\httplib2__init__.py", line 1139, in _conn_request
    raise ServerNotFoundError("Unable to find the server at %s" % conn.host)
httplib2.ServerNotFoundError: Unable to find the server at github.com

**Comment by macboy80** [comment]
If I click on :
https://github.com/c00w/btcnet_info/zipball/master
it just says "Not Found"

EDIT: That's because the github included the '
Nevermind...

**Comment by c00w** [comment]
@macboy80 Is it working now?

**Comment by macboy80** [comment]
Ok. So after several rapid attempts, it pulled the download and is working now. There are lots of errors in the logfile. I'm not sure if they are normal.

http://pastebin.com/PiUbuyy6

**Comment by c00w** [comment]
... Was your internet down? Everything was failing. If it wasn't you may want to try and update gevent, httplib2 and pythongit.

**Comment by macboy80** [comment]
My internet was up, and I was getting the above error. It took several attempts for the zip to download, however subsequent starts have been good so far. All prerequisites are up to date as well. As of right now bH is open and accepting shares. There are still some lagging issues that look to be between DB, MTRED, ELIGUIS, and DONATION and the above logfile output errors from the pastebin. I'm more than happy to continue to debug the remaining errors, but this issue appears to be solved at the moment.

EDIT: Just went to restart bH to delag, and took 3 tries to open. That is three tries to download the .zip

**Comment by c00w** [comment]
I recomend you remove your ipv6 interface. For some reason you are trying to mine with eligius over ipv6 and gevent doesn't support that cleanly.

**Comment by c00w** [comment]
You can delete the .zip and it will try to use the git interface.

**Comment by c00w** [comment]
Are you on windows or linux?

**Comment by macboy80** [comment]
I am on Win7 x64. I still get the original error referenced at the top of this issue if there is no zip in the bH directory. It will then go to fallback to download the zip which works most of the time. There are times when it takes a few tries to download the zip before starting. Also the zip in my bH directory is just called ".zip" There is no filename, the size shows as 1K, and winRAR says unknown or damaged.

**Comment by Clipseza** [comment]
I dont think you should be concerned with this issue, windows is bugging out all over the place since move to gevent and I think c0w got to narrow down on that issue first.

ie. stick to the version before change to gevent if you run on windows.

**Comment by macboy80** [comment]
@Clipseza Absolutely. I intend to do that on my server, however I am willing to help with bleeding edge whenever needed.
@c00w No hurry... I am happily hopping as well as I need to... Just let me know when you need some testing.

**Comment by c00w** [comment]
Um the .zip in the directory is just a flag telling the program that for whatever reason git isn't working and we should fallback to zip downloads. Its not actually a zip file.

**Comment by joulesbeef** [comment]
after the latest commit "windows fixes"

D:\Users\joulesbeef\Desktop\currentminer\bitHopper>rem With 2>> the program (errors) output is saved at the end of file without

D:\Users\joulesbeef\Desktop\currentminer\bitHopper>rem  overwrite it each time you open the program

D:\Users\joulesbeef\Desktop\currentminer\bitHopper>python bitHopper.py
22:46:56|btcnet_wrapper: Downloading zip archive
22:46:57|btcnet_wrapper: Extracting zip archive
22:46:57|btcnet_wrapper: Done
22:46:59|database: Cleaning Database
22:46:59|pool: pools-custom.cfg not found.
Traceback (most recent call last):
  File "D:\Python27\lib\site-packages\gevent\greenlet.py", line 402, in run
    result = self._run(_self.args, *_self.kwargs)
  File "D:\Users\joulesbeef\Desktop\currentminer\bitHopper\exchange.py", line 47, in update_profitability
    self.calc_profit()
  File "D:\Users\joulesbeef\Desktop\currentminer\bitHopper\exchange.py", line 33, in calc_profit
    btc_diff = self.difficulty.diff['btc']
KeyError: 'btc'
<Greenlet at 0x26598a0: <bound method Exchange.update_profitability of <exchange.Exchange instance at 0x0265EEB8>>> failed with KeyError

22:47:00|database: Cleaning Database
22:47:00|lp: LP Module Load
Traceback (most recent call last):
  File "bitHopper.py", line 317, in <module>
    main()
  File "bitHopper.py", line 237, in main
    bithopper_instance = BitHopper(options, config)
  File "bitHopper.py", line 86, in **init**
    self.workers = Workers.Workers(self)
  File "D:\Users\joulesbeef\Desktop\currentminer\bitHopper\Workers.py", line 17, in **init**
    self.fd = open('worker.cfg', 'rwb')
ValueError: Invalid mode ('rwb')

**Comment by macboy80** [comment]
So... An update. I just pulled the latest, and on the second startup, the downloading of the zip failed. That did not crash the program. It just displayed the error and kept running.

I would say this issue is solved on my end.

**Comment by c00w** [comment]
Sweet. I'm going to close this issue as it turned into a general support thread.

**Comment by macboy80** [comment]
I hate being the bearer of bad news, BUT I just tried to update my bH server which is WinXP x32. It seems this error still affects WinXP. It will not download the zip, but it will start. The lagging issue is still present as well, but I'm not sure this issue has anything to do with it. All software was updated to latest on my server.

@c00w You seemed so excited to close this issue, so I don't want to be the one to reopen it.
